### ISSUE TITLE
Disable logging state messages by default?

### ISSUE THREAD

**Issue by edgao** [airbytehq/airbyte#16033]
We don't really have control over what a connector puts in its state blob, so there's no safeguard against users putting sensitive data in there. https://github.com/airbytehq/oncall/issues/504 was an example of this. There's no way around storing the state (since we need that to run syncs), but at the very least we can avoid putting it into logs.

**Comment by evantahler** [comment]
A followup question:

The platform isn't logging the content of state messages normally... are you asking us to redact things in the log after the fact based on the content of state messages? Or, are you proposing that any content that sort-of looks like a state message should not be logged?

Or, is this something I broke recently as part of https://github.com/airbytehq/airbyte/pull/15895?

**Comment by edgao** [comment]
This is (I think, @git-phu correct me if I"m wrong) this is the log statement in question https://github.com/airbytehq/airbyte/blob/6991452ce5e25586d815fb91e5775da3b8b4e561/airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java#L431

which is unrelated to your change afaict. The immediate fix here is probably to add a flag to enable/disable that log statement (and make sure it's disabled by default + in cloud), though there might be other places where we log the state object.

Post hoc redaction / inline redaction would be interesting, and plausibly useful? But outside my intended scope :P

**Comment by git-phu** [comment]
@edgao yup, the line you linked appears to be the one we see in our logs.

Although there also appears to be an equivalent line for python sources coming from
https://github.com/airbytehq/airbyte/blob/05fdda7d7f2f5507162da99c1135c9e922ecccbb/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L220

**Comment by evantahler** [comment]
There are some other places in that same file where we log state:

https://github.com/airbytehq/airbyte/blob/6991452ce5e25586d815fb91e5775da3b8b4e561/airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java#L281-L288

I'm trying to ⚖️ the benefit of seeing the full state message to aid debugging vs the risk that the state message sometimes contain PII.  Can we re-visit the threat model of why logging state is bad?  Presumably the folks who have access to these logs are the same folks who have access to the source data, and they know the passwords and API Keys... This might not always be the case in the future I guess.  We do store potentially sensitive data in state messages (as discovered in the linked issue) in the platform database.

**Comment by evantahler** [comment]
cc @andyjih
